### PR TITLE
implemented LocalGitRepository for working with a local git folder

### PR DESCRIPTION
**Pull request by amirmalka**

### CODE REVIEW

File: core/pkg/resourcehandler/localgitrepository.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+func (g *LocalGitRepository) GetBranchName() string {
+	return g.head.Name().Short()
+}
+
+func (g *LocalGitRepository) GetOriginUrl() string {
+	branchName := g.GetBranchName()
+	if branchRef, branchFound := g.config.Branches[branchName]; branchFound {
+		remoteName := branchRef.Remote
+		return g.config.Remotes[remoteName].URLs[0]
```
Comment (dwertent) [inline core/pkg/resourcehandler/localgitrepository.go:62]: Are you sure we don't need to check the list length before accessing it?

File: core/pkg/resourcehandler/localgitrepository.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	head   *plumbing.Reference
+	config *config.Config
+}
+
+type GitCommit struct {
+	hash        string
+	authorName  string
+	authorEmail string
+	message     string
+	date        string
```
Comment (dwertent) [inline core/pkg/resourcehandler/localgitrepository.go:24]: Is it possible to use `time.Time`? What is the time format you can extract from the local git?

File: core/pkg/resourcehandler/localgitrepository.go (hunk 1 above)
Comment (amirmalka) [reply core/pkg/resourcehandler/localgitrepository.go:62]: So I followed the API documentation - they note that "it must be non-empty": https://github.com/go-git/go-git/blob/f0b111ab70e4e90013658b0835929b2083902017/config/config.go#L550

And I have seen that they also access the first URL without checking in other functions: https://github.com/go-git/go-git/blob/f0b111ab70e4e90013658b0835929b2083902017/config/config.go#L642

If you think we should check this on our side I will add a check

File: core/pkg/resourcehandler/localgitrepository.go (hunk 2 above)
Comment (amirmalka) [reply core/pkg/resourcehandler/localgitrepository.go:24]: Yes, `commit.Author.When` is of type `time.Time`. I will change it

**Review by dwertent — APPROVED** [review]

**Pull request merged**